homeassistant/components/assist_pipeline/pipeline.py

```diff
@@ -663,17 +819,45 @@ class PipelineInput:
     async def execute(self) -> None:
         """Run pipeline."""
         self.run.start()
-        current_stage = self.run.start_stage
+        current_stage: PipelineStage | None = self.run.start_stage
+        audio_buffer: list[bytes] = []
 
         try:
+            if current_stage == PipelineStage.WAKE:
+                assert self.stt_stream is not None
+                detect_result = await self.run.wake_word_detection(
+                    self.stt_stream, audio_buffer
+                )
+                if detect_result is None:
+                    # No wake word. Abort the rest of the pipeline.
+                    current_stage = None
```

Comment:
Can’t we just call it done. 
```suggestion
                    self.run.end()
                    return
```